woocommerce/woocommerce-android
# [Product AI v2] Photo section in preview + other changes

Closes: #11954 


### Description
This PR has the following changes:
1. Extract the full-screen image viewer to a reusable component, and updates it to use the [`Popup`](https://developer.android.com/reference/kotlin/androidx/compose/ui/window/package-summary#Popup(androidx.compose.ui.window.PopupPositionProvider,kotlin.Function0,androidx.compose.ui.window.PopupProperties,kotlin.Function0)) API.
2. Extract the selected Image row to a reusable component.
3. Update the UI of the preview screen to match the new design of v2.
4. Update the state modeling to allow handling a separate state for the Image.
5. Show the Image in the Preview screen.

### Testing information
1. Open the app.
2. Start AI product creation.
3. Add a photo.
4. Continue to the Preview screen.
5. Confirm the photo is shown according to the design.
6. Test the photo actions.
7. Confirm they work as expected.

### Images/gif
<img src="https://github.com/woocommerce/woocommerce-android/assets/1657201/18b8d58f-c7ac-4ff0-9f83-57ba345af8cc" width=360 />
